scripts/auto_labeling_using_llm.py

```diff
@@ -0,0 +1,286 @@
+    ASSETS_FOLDER,
+    CLASS_CONCEPTS_VALUES,
+    DATASET_NAME,
+    HF_TOKEN,
+    SPLITS,
+    CONCEPTS,
+    LABELED_CLASSES,
+)
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+def save_metadata(hf_api: HfApi, metadata: dict, split: str, push_to_hub: bool = False):
+    with jsonlines.open(f"{ASSETS_FOLDER}/{DATASET_NAME}/data/{split}/metadata.jsonl", mode="w") as writer:
+        writer.write_all(metadata)
+
+    if push_to_hub:
+        hf_api.upload_file(
+            path_or_fileobj=os.path.join("{ASSETS_FOLDER}/{DATASET_NAME}/data/{split}", "metadata.jsonl"),
+            path_in_repo=f"data/{split}/metadata.jsonl",
+            repo_id=DATASET_NAME,
+            repo_type="dataset",
+        )
+
+
+def get_votes(hf_api: HfApi):
+    hf_api.snapshot_download(
+        local_dir=f"{ASSETS_FOLDER}/{DATASET_NAME}",
+        repo_id=DATASET_NAME,
+        repo_type="dataset",
+    )
+    metadata = {}
+    for split in SPLITS:
+        metadata[split] = []
+        with jsonlines.open(f"{ASSETS_FOLDER}/{DATASET_NAME}/data/{split}/metadata.jsonl") as reader:
+            for row in reader:
+                metadata[split].append(row)
+    votes = {}
+    for filename in os.listdir(f"{ASSETS_FOLDER}/{DATASET_NAME}/votes"):
+        with open(f"{ASSETS_FOLDER}/{DATASET_NAME}/votes/{filename}") as f:
+            key = filename.split(".")[0]
+            votes[key] = json.load(f)
+    return metadata, votes
+
+
+def get_pre_labeled_concepts(item: dict):
+    active_concepts = CLASS_CONCEPTS_VALUES[item["class"]]
+    return {c: c in active_concepts for c in CONCEPTS}
+
+
+def compute_concepts(votes):
+    vote_sum = {c: 0 for c in CONCEPTS}
+    for vote in votes.values():
+        for c in CONCEPTS:
+            if c not in vote:
+                continue
+            vote_sum[c] += 2 * vote[c] - 1
+    return {c: vote_sum[c] > 0 if vote_sum[c] != 0 else None for c in CONCEPTS}
+
+
+class OpenAIRequest:
+    def __init__(self, model: str="gpt-4o"):
+        self.client = AzureOpenAI(
+            api_version=os.environ["AZURE_OPENAI_API_VERSION"],
+            azure_deployment=os.environ["AZURE_OPENAI_CHAT_DEPLOYMENT_NAME"],
+        )
+        self.concepts = ",".join(CONCEPTS)
+
+    def __call__(self, item: dict, icl: dict, **kwargs):
+        """Send a request to the OpenAI API."""
+        message = [
+            {
+                "role": "system",
+                "content": [
+                    {
+                        "type": "text",
+                        "text": """\
+You are a helpful assistant that can help annotating images. Answer by giving the list of concepts you can see in the provided image.
+
+Given an image and its class, provide the concepts that are present in the image.
+
+You may choose from the following concepts only:
+{concepts}
+
+Provide the classification in the following JSON format:
+{{"red": True, "sphere": True, "stem": False, ...}}
+""".format(concepts=self.concepts)
```

Comment:
I would write smth like:
```suggestion
Given an image and its class, annotate the concepts' presence in the image using a JSON format.

The labels must be provided according to the following JSON schema:
{concept_schema}
""".format(concept_schema={"properties":{concept: {"type":"boolean"} for concept in self.concepts}})
```